extension/packagemanagerextension/serverextension/processhelper.py

```diff
@@ -0,0 +1,93 @@
+# pylint: disable=C0321
+
+import json
+import os
+import re
+import uuid
+import yaml
+
+from subprocess import check_output, CalledProcessError
+from packaging.version import parse
+
+from traitlets.config.configurable import LoggingConfigurable
+from traitlets import Dict
+
+MAX_LOG_OUTPUT = 6000
+
+CONDA_EXE = os.environ.get("CONDA_EXE", "conda")  # type: str
+
+# try to match lines of json
+JSONISH_RE = r'(^\s*["\{\}\[\],\d])|(["\}\}\[\],\d]\s*$)'
+
+
+class ProcessHelper(LoggingConfigurable):
```

Comment:
Are this functions used in more than one place?

Reply:
If not, these could be in the place where they are needed. For example, I mentioned that you could group all swanproject related functions (read, update, get the file path) in one SwanProject class.

Reply:
> Are this functions used in more than one place?

Yes